From: Bastian Germann <bage@linutronix.de>
Origin: upstream, https://github.com/sbabic/swupdate/commit/0c672866d4a9fe8f0808b5d8a0afb95c13c4c138
Date: Mon, 6 Jul 2020 14:27:04 +0200
Subject: mtd/ubi: Use include dir that upstream suggests

mtd-utils does not export its libmtd/libubi.  However, their internal use
suggests placing the header files directly in /usr/include.
The mtd/... namespace is used by kernel header files, so it is not a good
idea to place and expect the headers there.

Debian sid just made the libraries available without the mtd/ include prefix.
Applying this patch demands a corresponding change in meta-swupdate.

Move the lib{mtd,ubi}.h files out of the mtd/ include prefix.

Signed-off-by: Bastian Germann <bage@linutronix.de>
---
 .travis.yml     | 4 ++--
 include/flash.h | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index d636a15..e1ebc1c 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -58,8 +58,8 @@ before_install:
     - ./autogen.sh
     - ./configure
     - make
-    - sudo install -m 644 include/libubi.h /usr/local/include/mtd
-    - sudo install -m 644 include/libmtd.h /usr/local/include/mtd
+    - sudo install -m 644 include/libubi.h /usr/local/include
+    - sudo install -m 644 include/libmtd.h /usr/local/include
     - sudo install -m 644 include/mtd/ubi-media.h /usr/local/include/mtd
     - sudo install -m 644 *.a /usr/local/lib
     - cd ..
diff --git a/include/flash.h b/include/flash.h
index 2999c8d..d3c1860 100644
--- a/include/flash.h
+++ b/include/flash.h
@@ -10,8 +10,8 @@
 #define _FLASH_PART_H
 
 #include <stdint.h>
-#include <mtd/libmtd.h>
-#include <mtd/libubi.h>
+#include <libmtd.h>
+#include <libubi.h>
 #include "bsdqueue.h"
 
 #define DEFAULT_CTRL_DEV "/dev/ubi_ctrl"
